### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by Ewoskyx**
###Add  Devise

In this PR, I did;
- Remove the current_user method in ApplicationController, devise will provide us with one.
- Install and setup devise
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed passwords should be stored in the database.
- Ask for confirmation of email.
- Can reset password.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
```
Comment (GzimAsani) [inline app/models/user.rb:2]: - [x] You have a typo here. You have written `:database_authenticatable` where it should be `:database_authenticable`

**Review by GzimAsani — CHANGES_REQUESTED** [review]
Hi @Ewoskyx   ,
 
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Your code organization is good.
 
 ## Required Changes 
 * [x]  The MIT link is broken. Kindly fix it. Your README is the entry point of your users and having it well written without any broken links gives a good first impression and maximizes the potential of your product to your users.
 
 _Check the comments under the review._
 
 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Ewoskyx) [reply app/models/user.rb:2]: Hello thank you for the review, I did check it once again, and the correct one is "authenticatable". Even in different frameworks like Laravel, it is the same.
Here are some sources: 
https://www.yourdictionary.com/authenticatable
https://en.wiktionary.org/wiki/authenticatable

Just to make sure, I did try your suggestion, but this returned with a name error.
Also, thank you once again for your time :)

Comment (GzimAsani) [reply app/models/user.rb:2]: Hei @Ewoskyx, I just double checked and you are absolutely right. Sorry for the confusion

**Review by andersonlebon — COMMENTED** [review]
## Hi @Ewoskyx 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is now professional :heavy_check_mark: 

## Optional suggestions
- [ ] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**